app/src/androidTest/java/ch/epfl/sdp/peakar/general/SettingsMapActivityTest.java

```diff
@@ -181,4 +193,39 @@ private void readFromFile() {
 
     }
 
+    /*Test switch between normal map and satellite view*/
+    @Test
+    public void pressSatelliteNormalMapButton() {
+        OSMMap osmMap = SettingsMapActivity.osmMap;
+        //Originally the map is set to default
+        //A press on the button will change the map tile for satellite
+
+        MapView mapView = osmMap.getMapView();
+        MapTileProviderBase tileProviderBase = mapView.getTileProvider();
+        assertEquals("Mapnik", tileProviderBase.getTileSource().name());
+        //Click on button
+        ViewInteraction button = onView(withId(R.id.changeMapTile));
+        button.perform(click());
+        //Check that the provider has changed
+        tileProviderBase = mapView.getTileProvider();
+        assertEquals("ARCGisOnline", tileProviderBase.getTileSource().name());
+    }
+
+    /* Check that a press on "zoom on user location button effectively zooms on user loc"*/
+    @Test
+    public void pressZoomOnUserLocationButton() throws InterruptedException {
+        OSMMap osmMap = SettingsMapActivity.osmMap;
+        MapView mapView = osmMap.getMapView();
+        //Get map center at start
+        GeoPoint geoPointStart = (GeoPoint) mapView.getMapCenter();
+        //Wait 5 Sec for the provider to get the location
+        Thread.sleep(5000);
```

Comment:
Use thread sleep constant instead. (the suggested constant needs to be added + a reference to TestingConstant file in the imports)
```suggestion
        Thread.sleep(THREAD_SLEEP_5S);
```